ZukoAzula2016/src/org/team1540/zukoazula/IntakeArm.java

```diff
@@ -1,22 +1,69 @@
 package org.team1540.zukoazula;
 
+import ccre.behaviors.ArbitratedFloat;
+import ccre.behaviors.Behavior;
+import ccre.behaviors.BehaviorArbitrator;
+import ccre.channel.BooleanCell;
 import ccre.channel.BooleanIO;
 import ccre.channel.BooleanInput;
 import ccre.channel.EventInput;
+import ccre.channel.EventLogger;
+import ccre.channel.EventOutput;
+import ccre.channel.FloatCell;
+import ccre.channel.FloatIO;
 import ccre.channel.FloatInput;
 import ccre.channel.FloatOutput;
+import ccre.cluck.Cluck;
 import ccre.ctrl.ExtendedMotor;
 import ccre.ctrl.ExtendedMotorFailureException;
 import ccre.drivers.ctre.talon.TalonExtendedMotor;
 import ccre.frc.FRC;
+import ccre.instinct.InstinctModule;
+import ccre.log.LogLevel;
+import ccre.timers.PauseTimer;
 
 public class IntakeArm {
     private static final TalonExtendedMotor intakeArmCAN = FRC.talonCAN(9);
-    private static final FloatInput encoder = intakeArmCAN.modEncoder().getEncoderPosition();
+    private static final FloatIO encoder = intakeArmCAN.modEncoder().getEncoderPosition();
+    private static final FloatInput outputCurrent = intakeArmCAN.modFeedback().getOutputCurrent();
 
-    private static final FloatInput intakeArmAxis = ZukoAzula.controlBinding.addFloat("Intake Arm Axis").deadzone(0.2f).negated();
+    private static final FloatInput intakeArmAxis = ZukoAzula.controlBinding.addFloat("Intake Arm Axis").deadzone(0.2f).negated().multipliedBy(ZukoAzula.mainTuning.getFloat("Intake Arm Speed", .5f));
+
+    private static final BooleanCell needsToCalibrate = new BooleanCell(true);
+
+    private static final FloatCell armHigh = ZukoAzula.mainTuning.getFloat("Intake Arm High Position", 1f);
+
+    private static final BehaviorArbitrator armBehaviors = new BehaviorArbitrator("Intake Arm Behaviors");
+    private static final ArbitratedFloat control = armBehaviors.addFloat();
 
     public static void setup() throws ExtendedMotorFailureException {
-        intakeArmAxis.send(intakeArmCAN.simpleControl());
+        FloatInput armPosition = encoder.normalize(armHigh.minus(ZukoAzula.mainTuning.getFloat("Intake Distance Between Encoders", 2760)), armHigh.minus(ZukoAzula.mainTuning.getFloat("Intake Arm High Position Offset", 100)));
```

Comment:
It's not clear what all of these different tuning values for the arm location mean. Can you clarify them?